RecoLocalTracker/SiPhase2VectorHitBuilder/src/VectorHitBuilderAlgorithm.cc

```diff
@@ -0,0 +1,404 @@
+#include "RecoLocalTracker/SiPhase2VectorHitBuilder/interface/VectorHitBuilderAlgorithm.h"
+#include "Geometry/CommonDetUnit/interface/GeomDet.h"
+#include "Geometry/CommonTopologies/interface/PixelTopology.h"
+#include "DataFormats/TrackerRecHit2D/interface/VectorHit2D.h"
+
+bool VectorHitBuilderAlgorithm::LocalPositionSort::operator()(Phase2TrackerCluster1DRef clus1,
```

Comment:
way too expensive.
cache those position, please

Reply:
To better understand the comment, the idea is to create a map of clusters and x-positions to be passed to this operator so that it doesn't have to get the cluster parameters from the CPE for each comparison?

Reply:
I implemented a caching of the these comparisons now. @VinInn could you check if that is what you had in mind?

Reply:
well I had in mind a simple vector caching just cpe_->localParameters(*clus, *gdu);
one to one with the vector holding the clusters (YES, the clusters ARE stored in a trivial std::vector)
I suspect a full combinatorics map is why too expensive....

Reply:
As I got rid of the sorting, I removed all of this.